modules/c++/sys/source/OSUnix.cpp

```diff
@@ -189,9 +189,14 @@ std::string sys::OSUnix::getTempName(const std::string& path,
     else
     {
         name = tempname;
+        sys::File newFile(name, sys::File::READ_AND_WRITE, sys::File::CREATE);
         free(tempname);
```

Comment:
Unrelated but this is bugging me... in the `#if` section, we're using a fixed size buffer and some old school `strcat`s.  Can you change this to just use a `std::string`?